html/semantics/embedded-content/the-iframe-element/iframe_sandbox_allow_top_navigation.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Check that sandboxed iframe can perform navigation on the top frame
+           when allow-top-navigation is set</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <iframe sandbox="allow-top-navigation allow-scripts"></iframe>
+    <script>
+      if (opener) {
+        // We're the popup (i.e. a top frame).  Load into the iframe the page
+        // trying to modifying the top frame and transmit the result to our
+        // opener.
+        onmessage = function(e) {
+          opener.postMessage({data: e.data, origin: e.origin}, "*")
+          window.close();
+        }
+        document.querySelector("iframe").src = "support/iframe-that-performs-top-navigation-without-user-gesture-passed.html";
```

Comment:
It seems like the timing here's not going to work. The frame loads `iframe-that-performs-top-navigation-without-user-gesture-passed.html`, which navigates the page containing the frame. Depending on when that navigation fires, there might or might not actually be a message handler to `postMessage()` up to the `opener`.